### ISSUE TITLE
Waybar Crashes on Suspend

### ISSUE THREAD

**Issue by portalbyte** [Alexays/Waybar#3424]
Currently on Arch Linux. When I suspend both monitors go off and my screen locks. When I turn it back on, Waybar has crashed. The console often displays different messages, but they all end the same - that my config has an error in it. Yet my config is fine. What is happening?

This example has a HEADLESS-1 error, but this doesn't always show up.

```
[2024-07-08 17:03:02.696] [debug] Cmd exited with code 0
[2024-07-08 17:03:02.696] [debug] Received SIGCHLD in signalThread
[2024-07-08 17:03:02.696] [debug] Received SIGCHLD in signalThread
[2024-07-08 17:03:02.697] [debug] Cmd exited with code 0
[2024-07-08 17:03:02.701] [debug] Cmd exited with code 0
[2024-07-08 17:03:02.701] [debug] Received SIGCHLD in signalThread
[2024-07-08 17:03:02.702] [debug] Received SIGCHLD in signalThread
[2024-07-08 17:03:02.702] [debug] Cmd exited with code 0
[2024-07-08 17:03:02.820] [debug] hyprland IPC received monitorremoved>>DP-1
[2024-07-08 17:03:02.821] [debug] hyprland IPC received activewindow>>LibreWolf,New Issue · Alexays/Waybar — LibreWolf
[2024-07-08 17:03:02.944] [debug] Output removed: HP Inc. HP 24fw
[2024-07-08 17:03:02.948] [info] Bar removed from output: DP-1
[2024-07-08 17:03:02.948] [debug] hyprland IPC received activewindowv2>>5960b9723fb0
[2024-07-08 17:03:02.949] [debug] hyprland IPC received workspace>>1
[2024-07-08 17:03:02.950] [debug] hyprland IPC received workspacev2>>1,1
[2024-07-08 17:03:02.950] [trace] Getting active workspaces
[2024-07-08 17:03:02.951] [debug] hyprland IPC received createworkspace>>11
[2024-07-08 17:03:03.060] [trace] Updating workspace states
[2024-07-08 17:03:03.065] [trace] Updating window count
[2024-07-08 17:03:03.065] [debug] Workspace created: 11
[2024-07-08 17:03:03.065] [debug] Output removed: HP Inc. HP 24fw
basic_string::_M_create
[2024-07-08 17:03:03.065] [debug] Output detection done: HEADLESS-1 ()
[2024-07-08 17:03:03.065] [trace] Extending orphans with workspace 2
[2024-07-08 17:03:03.065] [trace] Extending orphans with workspace 10
[2024-07-08 17:03:03.065] [trace] Extending orphans with workspace 4
[2024-07-08 17:03:03.065] [trace] Extending orphans with workspace 3
[2024-07-08 17:03:03.065] [trace] Extending orphans with workspace 1
[2024-07-08 17:03:03.065] [trace] Extending orphans with workspace 9
[2024-07-08 17:03:03.065] [trace] Extending orphans with workspace 20
[2024-07-08 17:03:03.065] [trace] Extending orphans with workspace 5
[2024-07-08 17:03:03.065] [debug] window-rewrite is not defined or is not an object, using default rules.
[2024-07-08 17:03:03.065] [error] Monitor 'HEADLESS-1' does not have an ID? Using 0
[2024-07-08 17:03:03.066] [debug] hyprland IPC received createworkspacev2>>11,11
[2024-07-08 17:03:03.066] [warning] module hyprland/workspaces: Disabling module "hyprland/workspaces", Error parsing JSON: * Line 1, Column 1
  Syntax error: value, object or array expected.

[2024-07-08 17:03:03.066] [warning] module hyprland/window: Disabling module "hyprland/window", Error parsing JSON: * Line 1, Column 1
  Syntax error: value, object or array expected.

[2024-07-08 17:03:03.068] [debug] hyprland IPC received activewindow>>,
terminate called after throwing an instance of 'std::runtime_error'
  what():  Error parsing JSON: * Line 1, Column 1
  Syntax error: value, object or array expected.

[1]    108828 IOT instruction (core dumped)  waybar -l trace
```

OS: Arch Linux
Version: Waybar v0.10.3

**Comment by RobertMueller2** [comment]
The error message is referring to json content from hyprland IPC, not json in the config. 

I think this might have been fixed by #3389. Any chance you can try git version of Waybar?

**Comment by portalbyte** [comment]
Just tried the -git version, no issues! Appears to have been fixed.
